CMakeLists.txt

```diff
@@ -526,6 +535,10 @@ if(OS_FREEBSD OR OS_MACOS)
         set(HAVE_BUILTIN_ATOMICS True)
 endif()
 
+if (ENABLE_H2O AND NOT ARCH_SUPPORTS_64BIT_ATOMICS)
+       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DH2O_NO_64BIT_ATOMICS")
+endif (NOT ARCH_SUPPORTS_64BIT_ATOMICS)
+
```

Comment:
- This should be down with the other H2O related compliation stuff (around line 1650) so that it’s clear that it’s tied to H2O itself, not to our own H2O support.
- This needs to be added using `target_compile_options()` or better yet `target_compile_definitions()` for the `h2o` target so that it only affects that target and things which depend on that target, as well as being properly applied regardless of whether the thing being compiled is C or C++. `${CMAKE_C_FLAGS}` only affects C compiler invocations, and it will bleed over into essentially every external project we use as well which is undesirable for multiple reasons.

Reply:
@Ferroin thanks for review, I have addressed your feedback  and updated the PR